protocol/communities/permission_checker.go

```diff
@@ -33,6 +33,19 @@ type DefaultPermissionChecker struct {
 	logger *zap.Logger
 }
 
+type PreParsedPermissionsData struct {
+	Erc20TokenRequirements  map[uint64]map[string]*protobuf.TokenCriteria
+	Erc721TokenRequirements map[uint64]map[string]*protobuf.TokenCriteria
+	Erc20TokenAddresses     []gethcommon.Address
+	Erc20ChainIDsMap        map[uint64]bool
```

Comment:
```suggestion
	Erc20ChainIDsMap        map[uint64]struct{}
```

It is more idiomatic in go to use `map[uint64]struct{}` for sets because it clearly indicates that only the presence of keys matters.